packages/signature-controller/src/SignatureController.ts

```diff
@@ -406,6 +426,45 @@ export class SignatureController extends BaseControllerV2<
     }
   }
 
+  /**
+   * Called when MMI needs to set the defer value in the message.
+   *
+   * @param messageId - The id of the Message to update.
+   * @param deferAsSigned - The defer value to add in the message params.
+   */
+  setDeferSetAsSigned(messageId: string, deferAsSigned: any) {
```

Comment:
How? The reason why we added that method was because when a signature request is created and enters for example in the `newUnsignedTypesMessage` it didn't bring the `deferSetAsSigned` value in the `messageParams`. So in the extension we are calling this method `setDeferSetAsSigned` in order to update the message with that property when the signature request comes from a custodian.